python/tank/descriptor/descriptor_bundle.py

```diff
@@ -56,6 +60,124 @@ def version_constraints(self):
 
         return constraints
 
+    _sg_studio_versions = {}
+
+    @classmethod
+    def _get_sg_version(cls, connection):
+        """
+        Returns the version of the studio shotgun server. It caches the result per site.
+
+        :param connection: Connection to the Shotgun site.
+        :type: :class:`shotgun_api3.Shotgun`
+
+        :returns: a string on the form "X.Y.Z"
+        :rtype: str
+        """
+        if connection.base_url not in cls._sg_studio_versions:
+            try:
+                version_tuple = connection.server_info["version"]
+            except Exception, e:
+                raise TankDescriptorError("Could not extract version number for site: %s" % e)
+            cls._sg_studio_versions[connection.base_url] = ".".join([str(x) for x in version_tuple])
+
+        return cls._sg_studio_versions[connection.base_url]
+
+    def _test_constraint(self, key, current_version, item_name, reasons):
+        """
+        Tests a constraint and ensures the user provided a value and that the constraint is matched.
+
+        :param key: Name of the version constraint
+        :param current_version: Version the user passed in.
+        :param item_name: Pretty name for the version constraint.
+        :param reasons: List of reasons errors will be added to.
+
+        :returns: ``True`` if the constraint test passed, ``False`` if not.
+        """
+
+        constraints = self.version_constraints
+        if key in constraints:
+            minimum_version = constraints[key]
+            if not current_version:
+                reasons.append("Requires at least %s %s but no version was specified." % (item_name, minimum_version))
+                return False
+            if is_version_older(current_version, minimum_version):
+                reasons.append("Requires at least %s %s but currently installed version is %s." % (
+                    item_name, minimum_version, current_version
+                ))
+                return False
+        return True
+
+    def check_version_constraints(
+        self,
+        connection=None,
+        core_version=None,
```

Comment:
Not unless we add a utility method that allows to retrieve the descriptor for the current core.